src/cell_lists/dictionary.jl

```diff
@@ -25,6 +25,8 @@ struct DictionaryCellList{NDIMS} <: AbstractCellList
     end
 end
 
+supported_update_strategies(::DictionaryCellList) = (:semi_parallel, :serial)
```

Comment:
Why do you return tuples of symbols instead of distinct values that can be used for type-stable dispatch? E.g., along the lines of
```julia
struct UpdateStrategy{T} end
const semi_parallel = UpdateStrategy{:semi_parallel}()
const serial = UpdateStrategy{:serial}()

...

foo(::UpdateStrategy{:semi_parallel}) = "only semi_parallel"
bar(::Union{UpdateStrategy{:semi_parallel}, UpdateStrategy{:serial}}) = "serial or semi_parallel"

foo(semi_parallel)
bar(serial)
```
?

Of course, still observe YAGNI, so maybe the current approach is sufficient

Reply:
How about
```julia
struct ParallelUpdate end
struct SemiParallelUpdate end
struct SerialUpdate

struct NeighborhoodSearch{U}
    update_strategy :: U
end

supported_update_strategies() = (ParallelUpdate(), SemiParallelUpdate(), SerialUpdate())

update!(nhs) = update!(nhs, nhs.update_strategy)
update!(nhs, ::ParallelUpdate) = ...
update!(nhs, ::Union{SerialUpdate, SemiParallelUpdate}) = ...
```
and then
```julia
nhs = NeighborhoodSearch(update_strategy = SerialUpdate())
```

Reply:
Yes, much better!